analytics/analytics/job_processor/dimensions.py

```diff
@@ -29,6 +32,121 @@ def __init__(self, job_id: int, name: str) -> None:
         super().__init__(message)
 
 
+@dataclass(frozen=True)
+class RetryInfo:
+    is_retry: bool
+    is_manual_retry: bool
+    attempt_number: int
+    final_attempt: bool
+
+
+def _job_retry_data(
```

Comment:
I think this function is a bit misplaced here, since it's used in multiple files, and it feels off to lump it into "dimensions". Maybe it could go in `utils.py`, or even a new file.